### PR TITLE
external_deps: allow to fetch archives from various mirrors

### LINKED ISSUE #1006: `openal-soft.org` website is down
I noticed it yesterday, https://openal-soft.org doesn't respond (but ping does).

This prevents to fetch the external deps in the docker script to build releases.

Issue is tracked on OpenAL side there:

- https://github.com/kcat/openal-soft/issues/956

I don't know if there are mirrors, if there are, it may be useful to update our script to iterate known mirrors in case the primary source is not available.

### PR DESCRIPTION
**Pull request by illwieckz**
Allow to fetch archives from various mirrors, cycling until one works.

Add a bunch of known mirrors for some packages.

Do some minor rewriting to unify the wording.

Fixes #1006 and #1007:

- https://github.com/DaemonEngine/Daemon/issues/1006
- https://github.com/DaemonEngine/Daemon/issues/1007

### CODE REVIEW

File: external_deps/build.sh [hunk 1]
```diff
@@ -59,54 +59,65 @@ LDFLAGS=''
...
-	fi
-	extract "${1}" "${3}"
+	local extract_dir="${BUILD_DIR}/${1}"; shift
+	local tarball_file="${DOWNLOAD_DIR}/${1}"; shift
+	while [ ! -f "${tarball_file}" ]; do
+		[ -n "${1:-}" ]
+		local download_url="${1}"; shift
+		echo "Downloading ${download_url}"
+		"${CURL}" -L --fail -o "${tarball_file}" "${download_url}" \
+		|| rm -f "${tarball_file}"
```
Comment (DolceTriade) [inline external_deps/build.sh:107]: you should log on error at least

File: external_deps/build.sh [hunk 2]
```diff
@@ -59,54 +59,65 @@ LDFLAGS=''
...
 # Usage: download <filename> <URL> <dir>
 download() {
-	if [ ! -f "${DOWNLOAD_DIR}/${1}" ]; then
-		"${CURL}" -L --fail -o "${DOWNLOAD_DIR}/${1}" "${2}"
-	fi
-	extract "${1}" "${3}"
+	local extract_dir="${BUILD_DIR}/${1}"; shift
+	local tarball_file="${DOWNLOAD_DIR}/${1}"; shift
+	while [ ! -f "${tarball_file}" ]; do
+		[ -n "${1:-}" ]
```
Comment (DolceTriade) [inline external_deps/build.sh:103]: Should we do something like:
`[ -n "${1:-}" ] ||( echo "No more URLs passed in!" >&2 && exit -1)`

**Review by DolceTriade — APPROVED** [review]

File: external_deps/build.sh (hunk 1 above)
Comment (illwieckz) [reply external_deps/build.sh:107]: Yeah I can add more logging, those two things you stumbled on are mostly code golf with implicit behaviour. 😁️
Even the `rm` is less about removing the fail (but curl says `--fail` may not remove the output on some errors, so we better want to make sure we remove it) than to turn the error into a `true` thanks to the `||`. 😅️

I'll make the code more explicit and more verbose.

**Comment by illwieckz** [comment]
@DolceTriade I added a commit to rework the logging of the script. So this change is larger than just applying your juggestions, can you review it as well? Thanks!

**Comment by illwieckz** [comment]
Well, the extra change is just logging and it works, so let's go.

**Comment by DolceTriade** [comment]
lgtm

**Pull request merged**